webapp-django/crashstats/home/views.py

```diff
@@ -1,10 +1,33 @@
+import functools
+
 from django.conf import settings
+from django.http import Http404
 from django.shortcuts import render
 from django.views.generic.base import RedirectView
 
 from crashstats.crashstats.decorators import pass_default_context
+from crashstats.crashstats.utils import build_default_context
+
+
+def handle_missing_product(view):
```

Comment:
Why only fix this for the homepage (or, rather, only views using this decorator), instead of for any view using `pass_default_context`?

Reply:
The bug (and the problem I wanted to tackle) is specifically about a fresh Socorro instance missing a product. I think you're likely to hit that when going to `/home/product/XXX` and not other places.

`pass_default_context` adds information to the context to handle the nav bar more than anything else. I think I'd rather those pages continue to do what they're doing now.

So for those reasons, I hedged towards just fixing the home view and not affecting anything else.

Relatedly, we've talked about redoing Socorro navigation and creating a real page that sits at `/` and lists the products available rather than automatically redirecting to `/home/product/XXX`. I think that's a more involved fix and something that's better to push off until after the photon overhaul. When we do that, we can undo the fix I did in this PR because it won't be helpful anymore.